kubernetes/services/rabbitmq.yml.j2

```diff
@@ -0,0 +1,76 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: rabbitmq
+spec:
+  replicas:             1
+  revisionHistoryLimit: 3
+  template:
+    metadata:
+      labels:
+        run: rabbitmq
+    spec:
+# The default name of the rabbitmq node is the hostname whose name in kubernetes is the name of the nodes.
+# This name is random and change after restarting, deleting, etc.
+# This behavior causes creates a new database for the new node, so we can not recover messages and queues.
+# We need a constant name of rabbitmq node, so we can read messages and queues from an existing database.
+# Adding hostname field in the kubernetes deployment, change host name of the pod.
```

Comment:
It's fine but please reword it:

The name of the RabbitMQ node (RABBITMQ_NODENAME) is by default the same as the hostname, which in Kubernetes is the same in every container running on a single Kubernetes node (and is the same as the hostname of that node). This name is random and changes when the pod is restarted or deleted and recreated. This behavior results in a new, empty Mnesia database being created when we recreate the pod and messages from the previous database not being recovered. For that reason we need to explicitly set the hostname for the pod and keep it unchanged between deployments.

Reply:
Done.